util/types/time.go

```diff
@@ -437,6 +437,15 @@ func (t *Time) check() error {
 	return nil
 }
 
+// TotalDays returns number of days starting from 0/1/1
+func (t Time) TotalDays() int {
```

Comment:
Or define a `DateDiff` which calls `calcDaynr` in `util/types` package, thus we can make sure its behavior is consistent with mysql ?  @Tratar 
Take a look at `TimeDiff` for example.